spec/abstract/SchedContext_A.thy

```diff
@@ -309,34 +309,40 @@ definition
   sched_context_update_consumed :: "obj_ref \<Rightarrow> (time,'z::state_ext) s_monad" where
   "sched_context_update_consumed sc_ptr \<equiv> do
     sc \<leftarrow> get_sched_context sc_ptr;
-    set_sc_obj_ref sc_consumed_update sc_ptr 0;
-    return (sc_consumed sc)
+    consumed <- return (sc_consumed sc);
+    if (consumed >= max_ticks_to_us)
+    then do
+        set_sc_obj_ref sc_consumed_update sc_ptr (consumed - max_ticks_to_us);
+        return max_ticks_to_us
+    od
+    else do
+        set_sc_obj_ref sc_consumed_update sc_ptr 0;
+        return (ticks_to_us consumed)
+    od
    od"
 
 definition
-  set_consumed :: "obj_ref \<Rightarrow> data list \<Rightarrow> (unit, 'z::state_ext) s_monad"
+  set_consumed :: "obj_ref \<Rightarrow> data option \<Rightarrow> (unit, 'z::state_ext) s_monad"
 where
-  "set_consumed sc_ptr args \<equiv>do
+  "set_consumed sc_ptr buf \<equiv> do
       consumed \<leftarrow> sched_context_update_consumed sc_ptr;
       ct \<leftarrow> gets cur_thread;
-      buffer \<leftarrow> return $ data_to_oref $ args ! 0;
-      sent \<leftarrow> set_mrs ct (Some buffer) ((ucast consumed) # [ucast (consumed >> 32)]);
+      sent \<leftarrow> set_mrs ct (map_option data_to_oref buf) ((ucast consumed) # [ucast (consumed >> 32)]);
```

Comment:
does this need to be here? I'm not sure if data should have been oref the whole time.